scripts/init-testnet.ts

```diff
@@ -0,0 +1,318 @@
+                borrowCap: "20000000000000000000000",
+                lendTokenId: {LendToken: 4} 
+            }
+        )
+    ];
+
+
+    let activateMarkets = [
+        api.tx.loans.activateMarket({Token: "KBTC"}),
+        api.tx.loans.activateMarket({Token: "KSM"}),
+        api.tx.loans.activateMarket({ForeignAsset: 1}),
+        api.tx.loans.activateMarket({ForeignAsset: 2}),
+    ];
+
+    return addMarkets.concat(activateMarkets);
+}
+
+function constructFundingSetup(api: ApiPromise) {
+    const tokens = [{Token: "KSM"}, {Token: "KINT"}];
+    const faucetSetup = tokens.map((token) => {
+        return api.tx.tokens.setBalance(
+            "5DqzGaydetDXGya818gyuHA7GAjEWRsQN6UWNKpvfgq2KyM7",
+            token,
+            20000000000000,
+            0
+        );
+    });
+    const calls = faucetSetup.concat([
+        api.tx.tokens.setBalance(
+            "a3cgeH7D28bBsHY4hGLzxkMFUcFQmjGgDa2kmxg3D9Z6AyhtL", // treasury
+            {Token: "KINT"},
+            "1000000000000000000000000", //1e12 KINT
+            0
+        )
+    ]);
+    return calls;
+}
+
+function constructVaultRegistrySetup(api: ApiPromise) {
+    const currencyPair = {
+        collateral: {ForeignAsset: 3}, // lksm 
+        wrapped: {Token: "KBTC"}
+    };
+    return [
+        api.tx.vaultRegistry.setLiquidationCollateralThreshold(currencyPair, "1450000000000000000"),
+        api.tx.vaultRegistry.setPremiumRedeemThreshold(currencyPair, "1650000000000000000"), 
+        api.tx.vaultRegistry.setSecureCollateralThreshold(currencyPair, "1800000000000000000"), 
+        api.tx.vaultRegistry.setMinimumCollateral(currencyPair.collateral, "20000000000000"), 
+        api.tx.vaultRegistry.setSystemCollateralCeiling(currencyPair, "38000000000000000"), 
+    ];
+}
+
+function constructRewardsSetup(api: ApiPromise) {
+    const blocksPerYears = 365 * 24 * 60 * 5; // 5 per minute
+    const vaultAnnuity = [
+        api.tx.tokens.setBalance(
+            "a3cgeH7D3w3wu37yHx4VZeae4EUqNTw5RobTp5KvcMsrPLWJg", // vaultAnnuity
+            {Token: "KINT"},
+            new BN(102803978514).muln(blocksPerYears),
+            0
+        ),
+        api.tx.vaultAnnuity.updateRewards(),
+    ];
+    const escrowAnnuity = [
+        api.tx.tokens.setBalance(
+            "a3cgeH7CzXoGgXh453eaSJRCvbbBKZN4mejwUVkic8efQUi5R", // escrowAnnuity
+            {Token: "KINT"},
+            new BN(47564687975).muln(blocksPerYears),
+            0
+        ),
+        api.tx.escrowAnnuity.updateRewards(),
+    ];
+    return vaultAnnuity.concat(escrowAnnuity);
+}
+
+function constructAmmSetup(api: ApiPromise) {
+    const pools = [
+        [{ Token: "KBTC" }, { Token: "KSM" }, 45_000],
+        [{ Token: "KBTC" }, { ForeignAsset: 1 }, 40_000], // usdt
+        [{ Token: "KBTC" }, { ForeignAsset: 2 }, 20_000], // movr
+        [{ Token: "KINT" }, { ForeignAsset: 2 }, 35_000], // movr
+    ];
+    const basicPoolSetup = pools.map(([token1, token2, reward]) => {
```

Comment:
You may need to explicitly indicate types for the tuple so TS knows which types to expect.
IIRC, it will think it could be any of the tuple's internal types, ie. treating the input as array of mixed types rather than a tuple.